tasmota/xdrv_41_tcp_bridge.ino

```diff
@@ -136,18 +328,21 @@ void TCPLoop(void)
 /********************************************************************************************/
 
 void TCPInit(void) {
+
   if (PinUsed(GPIO_TCP_RX) && PinUsed(GPIO_TCP_TX)) {
-    if (0 == (0x80 & Settings->tcp_config)) // !0x80 means unitialized
+    if (0 == (0x80 & Settings->tcp_config)) { // !0x80 means unitialized
       Settings->tcp_config = 0x80 | ParseSerialConfig("8N1"); // default as 8N1 for backward compatibility
+      if (!Settings->tcp_baudrate)  { Settings->tcp_baudrate = 115200 / 1200; } // only when uninitialized
+    }
     tcp_buf = (uint8_t*) malloc(TCP_BRIDGE_BUF_SIZE);
     if (!tcp_buf) { AddLog(LOG_LEVEL_ERROR, PSTR(D_LOG_TCP "could not allocate buffer")); return; }
+    serial_config_rfc2217 = Settings->tcp_config & 0x7F;
 
-    if (!Settings->tcp_baudrate)  { Settings->tcp_baudrate = 115200 / 1200; }
     TCPSerial = new TasmotaSerial(Pin(GPIO_TCP_RX), Pin(GPIO_TCP_TX), TasmotaGlobal.seriallog_level ? 1 : 2, 0, TCP_BRIDGE_BUF_SIZE);   // set a receive buffer of 256 bytes
-    TCPSerial->begin(Settings->tcp_baudrate * 1200, ConvertSerialConfig(0x7F & Settings->tcp_config));
+    TCPSerial->begin(Settings->tcp_baudrate * 1200 + (!Settings->tcp_baudrate ? 300 : 0), ConvertSerialConfig(0x7F & Settings->tcp_config));
```

Comment:
So Settings->tcp_baudrate == 0 means 300, 1 means 1200 etc ... ?

This would be clearer I think:
`Settings->tcp_baudrate ? Settings->tcp_baudrate * 1200 : 300`

Reply:
Absolutely right. Changed it in the second commit.